### PR TITLE
Introduce postgres

### PR DESCRIPTION
**Pull request by bakar-io**
- Replaces Redis with Postgres both as a vector store and as a general store (messages, assistants, etc.)
- Uses golang-migrate to manage migrations (use `make migrate`)
- Updates tests to use Postgres and adds a testing step to the CI

Storage methods have been rewritten to match the previous behavior that was present with Redis. After authentication is implemented, some of these methods will change (especially the handling of public assistants will need to change).

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -13,10 +16,9 @@ WORKDIR /backend
 # Copy only dependencies
 COPY pyproject.toml poetry.lock* ./
 
-# Install dependencies
-# --only main: Skip installing packages listed in the [tool.poetry.dev-dependencies] section
+# Install all dependencies
 RUN poetry config virtualenvs.create false \
-    && poetry install --no-interaction --no-ansi --only main
```
Comment (nfcampos) [inline backend/Dockerfile:19]: why is this changing?

File: backend/app/checkpoint.py [hunk 2]
```diff
@@ -55,31 +31,53 @@ def config_specs(self) -> list[ConfigurableFieldSpec]:
...
-            # unknown version
-            return None
+        if self.pg_pool is None:
+            self.pg_pool = get_pg_pool()
+
+        try:
+            async with self.pg_pool.acquire() as conn:
+                await conn.execute(
+                    """
+                    CREATE TABLE IF NOT EXISTS checkpoints (
```
Comment (nfcampos) [inline backend/app/checkpoint.py:45]: this should move to migrations?

Comment (bakar-io) [reply backend/app/checkpoint.py]: Good call. Done.

File: backend/Dockerfile (hunk 1 above)
Comment (bakar-io) [reply backend/Dockerfile:19]: So that pytest is available for those who develop inside a docker container and want to run tests inside.

**Pull request merged**